File: src/main/java/com/ciazhar/authserver/model/Photo.java
package com.ciazhar.authserver.model;



import javax.persistence.Entity;
import javax.persistence.Id;
import java.util.Date;

/**
 * Created by ciazhar on 6/21/17.
 */

@Entity
public class Photo {

    @Id
    private String id;
    private String accountId;
    private Date createdDate;
    private String url;

    public Photo(){
        this.setCreatedDate(new Date());
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAccountId() {
        return accountId;
    }

    public void setAccountId(String accountId) {
        this.accountId = accountId;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }
}

File: src/main/kotlin/com/ciazhar/authserver/model/jpa/Photo.kt
package com.ciazhar.authserver.model.jpa


import org.hibernate.annotations.GenericGenerator
import javax.persistence.Entity
import javax.persistence.Id
import java.util.Date
import javax.persistence.Column
import javax.persistence.GeneratedValue
import javax.validation.constraints.NotNull

/**
 * Created by ciazhar on 6/21/17.
 */

@Entity
data class Photo (

        @Id
        @GeneratedValue(generator = "uuid")
        @GenericGenerator(name = "uuid", strategy = "uuid2")
        @field:Column(name = "id_permission")
        var id: String? = null,

        @field:NotNull
        var accountId: String? = null,

        @field:NotNull
        var createdDate: Date? = null,

        @field:NotNull
        var url: String? = null
){
    init {
        this.createdDate = Date()
    }
}
